src/pages/index.js

```diff
@@ -86,18 +128,96 @@ function openModal(modal) {
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileNameElement.textContent = nameInput.value;
-  profileJobElement.textContent = jobInput.value;
-  closeModal(editProfileModal);
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+
+  api.editUserInfo({ name: nameInput.value, about: jobInput.value })
+  .then((data) => {
+    profileNameElement.textContent = data.name;
+    profileJobElement.textContent = data.about;
+    closeModal(editProfileModal);
+  })
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => {
+    setButtonText(submitButton, false);
+  });
 }
 
 function handleAddFormSubmit(evt) {
   evt.preventDefault();
-  const inputValues = { name: captionInput.value, link: linkInput.value };
-  const cardElement = getCardElement(inputValues);
-  cardsList.prepend(cardElement);
-  evt.target.reset();
-  closeModal(addImageModal);
+
+ const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+  api.addNewCard({ name: captionInput.value, link: linkInput.value })
+  .then((data) => {
+    const cardElement = getCardElement(data);
+    cardsList.prepend(cardElement);
+    evt.target.reset();
+    closeModal(addImageModal);
+  })
+  .catch((err) => {
+    console.error(err);
+  })
+  .finally(() => {
+    setButtonText(submitButton, false);
+  });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+ const submitButton = evt.submitter;
+  setButtonText(submitButton, true);
+    api.editAvatarImage(avatarLink.value)
+  .then((data) => {
+    profileImage.src = data.avatar;
+    evt.target.reset();
+    closeModal(avatarModal);
+  })
+  .catch((err) => {
+    console.error(err);
+  }).finally(() => {
+    setButtonText(submitButton, false);
+  });
+}
+
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  setButtonText(submitButton, true, "Delete", "Deleting...");
+  api.deleteCard(selectedCardId)
+  .then(() => {
+    selectedCard.remove();
+    closeModal(deleteCardModal);
+  })
+  .catch((err) => {
+    console.error(err);
+  }).finally(() => {
+    setButtonText(submitButton, true, "Delete", "Deleting...");
```

Comment:
`true`

In `.finally`, you should use `false`, to return the initial button text back